app/services/stats/r007_closed_cases_report.rb

```diff
@@ -0,0 +1,27 @@
+module Stats
+  class R007ClosedCasesReport < BaseCasesReport
+    def self.title
+      'Closed cases report'
+    end
+
+    def self.description
+      'Entire list of closed cases'
+    end
+
+    def case_scope
+      closed_scope = Case::Base.presented_as_closed
+
+      if @user.responder_only?
+        case_ids = Assignment.with_teams(@user.responding_teams).pluck(:case_id)
+        closed_scope.where(id: case_ids).most_recent_first
+      else
+        closed_scope.most_recent_first
+      end
+    end
+
```

Comment:
I'm probably missing something but I don't think we have test coverage for the `if` and the `else` branches here?

This report seems to need to know a lot about the world - would it make sense to move that `Assignment.with_teams...pluck(:case_id)` into a private method or a method on User?

Reply:
I have reused the `CaseFinderService` existing code rather than copying in the code.